src/main/java/com/aws/greengrass/logmanager/services/DiskSpaceManagementService.java

```diff
@@ -31,7 +31,13 @@ public List<String> freeDiskSpace(LogFileGroup group) {
 
         long bytesDeleted = 0;
         long minimumBytesToBeDeleted = Math.max(group.totalSizeInBytes() - group.getMaxBytes().get(), 0);
+        // Prefer to delete files we've already processed, however we do also need to clean up files that we
+        // have not processed as they may exceed the max disk space setting.
         List<LogFile> deletableFiles = group.getProcessedLogFiles();
+        deletableFiles.addAll(group.getLogFiles());
```

Comment:
lazy question, `group.getLogFiles` distinct from `group.getProcessedLogFiles`?   seems like we dont want duplicates in delteableFiles

Reply:
yes, they're distinct. Even if we duplicate it won't matter as deletion is safe if the file is deleted already.